src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        for (const key in action.extraData) {
+          state[key] = action.extraData[key];
+        }
+        break;
+
+      case action.type === 'removeProperties':
+        const actionKeysToRemove = action.keysToRemove;
+
+        for (let i = 0; i < actionKeysToRemove.length; i++) {
+          delete state[action.keysToRemove[i]];
+        }
```

Comment:
You can make it a bit simpler by using for of loop:
```suggestion
 for (const key of action.keysToRemove) {
          delete state[key];
 }
```